app/src/main/java/ryey/easer/skills/operation/http_request/HttpRequestOperationData.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2016 - 2019 Rui Zhao <renyuneyun@gmail.com>
+ *
+ * This file is part of Easer.
+ *
+ * Easer is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Easer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Easer.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ryey.easer.skills.operation.http_request;
+
+import android.os.Build;
+import android.os.Parcel;
+
+import org.jetbrains.annotations.NotNull;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.Serializable;
+import java.util.Objects;
+import java.util.Set;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import ryey.easer.Utils;
+import ryey.easer.commons.local_skill.IllegalStorageDataException;
+import ryey.easer.commons.local_skill.dynamics.SolidDynamicsAssignment;
+import ryey.easer.commons.local_skill.operationskill.OperationData;
+import ryey.easer.plugin.PluginDataFormat;
+
+public class HttpRequestOperationData implements OperationData, Serializable {
```

Comment:
Why implement `Serializable`...?

Reply:
Oh.  I just tested if parcel also works by using the Java serialization. Therefore I added Serializable, but I forgot to revert it again.  I'll remove it.  BTW it works and one could simplify the serialization methods a lot.  I might create another pull request for that after this merge, when I know Parcel a bit better.